### PR TITLE
[jsk_2016_01_baxter_apc] Added creative camera on baxter's rightarm

### PR DESCRIPTION
**Pull request by yuyu2172**
screenshot from 2016-04-06 21 55 47

### CODE REVIEW

**Comment by knorth55** [comment]
just to make sure. 
I don't know enough about urdf, so it can be stupid questions.

- did you check whether you can get tf of the camera?
- did you check tf view_frames that the camera is correctly embedded in system?
 
I'm not sure whether using urdf or using static_transform_publisher is better in real world experiment.

**Comment by knorth55** [comment]
also, can you move the camera to new position?

**Comment by pazeshun** [comment]
At least, this file should be xacro file because @yuyu2172 used xacro grammar in this file.  
In the case of Baxter, we used xacro file to describe robot model. When Baxter brings up, xacro is converted to urdf and that urdf is sent to ROS parameter server. The name of that parameter is `/robot_description`
https://github.com/start-jsk/jsk_apc/blob/master/jsk_2015_05_baxter_apc/launch/baxter.launch

File: jsk_2016_01_baxter_apc/urdf/baxter_creative.urdf [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    <parent link="left_wrist"/>
+    <child link="left_custom_vacuum_link"/>
+    <origin rpy="0 0 0" xyz="${left_wrist_offset_x} 0 0.19" />
+  </joint>
+  <joint name="left_custom_joint_base" type="fixed">
+    <parent link="left_wrist"/>
+    <child link="left_custom_vacuum_link_base"/>
+    <origin rpy="0 0 1.57" xyz="${left_wrist_offset_x} -0 0.13" />
+  </joint>
+  <!-- <<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<< custom vacuum gripper -->
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/urdf/baxter_creative.urdf]: line 14-99 is not necessary, no?

**Comment by yuyu2172** [comment]
@knorth55 
As @pazeshun commented, this is xacro file.

You can get tf of the camera from the model. Try a launch file below. 
```
<launch>
  <arg name="model" />
  <arg name="gui" default="False" />
  <arg name="rvizconfig" default="$(find jsk_2016_01_baxter_apc)/urdf.rviz" />
    <param name="robot_description" command="$(find xacro)/xacro.py $(find jsk_2016_01_baxter_apc)/urdf/baxter.xacro" />
  <param name="use_gui" value="$(arg gui)"/>
  <node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" />
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(arg rvizconfig)" required="true" />
</launch>
```

The position and orientation of the camera are determined by numbers in line 104.
I realized that the numbers became obsolete after updating the camera position. I will update them tomorrow. Please do not merge yet.

In addition, those numbers should be fine-tune through calibration in the future.

**Comment by yuyu2172** [comment]
@wkentaro 
baxter_creative.urdf can be used as an alternative to baxter.urdf in last year's directory.
Line 14-99 is necessary for the launch file in comment to run. 
This is not optimal.

**Comment by pazeshun** [comment]
@yuyu2172 
baxter.urdf doesn't exist in last year's directory, is it? baxter.xacro exists.
Maybe you should change the extension of the file to .xacro from .urdf.

**Comment by knorth55** [comment]
@yuyu2172 thanks. can you put a screenshot of tf_echo and view_frames?
just to make sure.

**Comment by yuyu2172** [comment]
@pazeshun 
right

@knorth55 
here.
screenshot from 2016-04-09 21 09 02

**Comment by knorth55** [comment]
@yuyu2172 thank you :)

**Comment by wkentaro** [comment]
@yuyu2172
I mean you can include jsk_2015_05_baxter_apc/urdf/baxter.xacro.

2016年4月9日土曜日、Yusuke Niitani<notifications@github.com>さんは書きました:

> @wkentaro <https://github.com/wkentaro>
> baxter_creative.urdf can be used as an alternative to baxter.urdf in last
> year's directory.
> Line 14-99 is necessary for the launch file in comment to run.
> This is not optimal.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/start-jsk/jsk_apc/pull/1240#issuecomment-207767344>
>


-- 
和田 健太郎 / Kentaro Wada
http://wkentaro.com

**Comment by wkentaro** [comment]
@yuyu2172 why not use name baxter.xacro?

**Comment by yuyu2172** [comment]
@wkentaro 
changed

**Comment by wkentaro** [comment]
@yuyu2172 The code LGTM.

Next, please check it works.

**Comment by yuyu2172** [comment]
Point clouds are now published.
screenshot from 2016-04-11 11 21 58

**Pull request merged**